core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java

```diff
@@ -214,7 +214,6 @@ private void addClassTypeHierarchy(CombinedIndexBuildItem combinedIndexBuildItem
                         .builder(name.toString())
                         .methods(true)
                         .fields(true)
-                        .finalFieldsWritable(doFinalFieldsNeedToBeWritable(info, finalFieldsWritable))
```

Comment:
Shouldn't we remove the `doFinalFieldsNeedToBeWritable()` method as well?

Reply:
Done